File: samples/JavascriptSamples/assets/js/samples/samples_common.js
import { Generator } from '../lib/salmon-core/salmon/generator.js';
import { Password } from '../lib/salmon-core/salmon/password/password.js';
import { MemoryStream } from '../lib/salmon-core/streams/memory_stream.js';

// create an encryption key from a text password
export async function getKeyFromPassword(password) {
	// generate a salt
	let salt = Generator.getSecureRandomBytes(24);
	// make sure the iterations are a large enough number
	let iterations =  60000;
    
	// generate a 256bit key from the text password
    let key = await Password.getKeyFromPassword(password, salt,iterations, 32);
	
    return key;
}

export async function generateRandomData(size) {
	let memoryStream = new MemoryStream();
	let buffer = new Uint8Array(65536);
	let len = 0;
	while(size > 0) {
		crypto.getRandomValues(buffer);
		len = Math.min(size, buffer.length);
		await memoryStream.write(buffer, 0, len);
		size -= len;
	}
	await memoryStream.flush();
	await memoryStream.close();
	return memoryStream.toArray();
}

File: samples/DotNetAndroidSamples/DotNetAndroidSamples/Samples/SamplesCommon.cs
using Mku.Streams;
using System;

namespace Mku.Salmon.Samples.Samples;

public class SamplesCommon
{
    static readonly Random random = new Random((int)Time.Time.CurrentTimeMillis());

    // create an encryption key from a text password
    public static byte[] GetKeyFromPassword(string password)
    {
        // generate a salt
        byte[] salt = Generator.GetSecureRandomBytes(24);
        // make sure the iterations are a large enough number
        int iterations = 60000;

        // generate a 256bit key from the text password
        byte[] key = Password.Password.GetKeyFromPassword(password, salt, iterations, 32);

        return key;
    }

    public static byte[] GenerateRandomData(int size)
    {
        MemoryStream memoryStream = new MemoryStream();
        byte[] buffer = new byte[65536];
        while (size > 0)
        {
            random.NextBytes(buffer);
            int len = Math.Min(size, buffer.Length);
            memoryStream.Write(buffer, 0, len);
            size -= len;
        }
        memoryStream.Flush();
        memoryStream.Close();
        return memoryStream.ToArray();
    }
}